exa_error.go

```diff
@@ -53,17 +53,32 @@ func (builder *ExaError) ParameterWithDescription(name string, value interface{}
 }
 
 func (builder *ExaError) Mitigation(mitigation string) *ExaError {
-	builder.mitigation = mitigation
+	builder.mitigations = append(builder.mitigations, mitigation)
 	return builder
 }
 
 func (builder *ExaError) String() string {
 	var stringBuilder strings.Builder
 	formattedMessage := formatMessage(builder)
 	stringBuilder.WriteString(fmt.Sprintf("%s: %s", builder.errorCode, formattedMessage))
-	if len(builder.mitigation) > 0 {
-		stringBuilder.WriteString(fmt.Sprintf(" %s", builder.mitigation))
+
+	if len(builder.mitigations) > 0 {
```

Comment:
How about using `strings.join`?
```golang
return strings.Join(mitigations[:], "\n* ")
```

Reply:
Wouldn't work since you'd miss the first "\n* " using strings.join